src/agent/tools/graph_search.py

```
@@ -0,0 +1,748 @@
+        entity_value_lower = entity_value.lower().strip()
+
+        # Search through all entities
+        candidates = []
+        for entity in self.knowledge_graph.entities:
+            if entity.confidence < min_confidence:
+                continue
+
+            # Check type match if specified
+            if entity_type:
+                # Handle both EntityType enum (entity.type) and string (mocks)
+                entity_type_value = entity.type if hasattr(entity.type, 'value') else entity.type
```

Ruff (0.14.5)

Useless `if`-`else` condition

(RUF034)